projects/vdk-core/src/vdk/api/plugin/connection_hook_spec.py

```diff
@@ -68,6 +72,70 @@ def db_connection_decorate_operation(
         """
         pass
 
+    @hookspec(firstresult=True)
+    def db_connection_execute_operation(
+        self, execution_cursor: ExecutionCursor
+    ) -> Optional[ExecuteOperationResult]:
+        """
+        The method that executes the actual SQL query using the raw_cursor.
+
+        The default implementation is basically this:
+        .. code-block::
+
+            if parameters:
+                return raw_cursor.execute(operation, parameters)
+            return raw_cursor.execute(operation)
+
+
+        Hook executions will stop at first hook implementation that returns non-None result.
+        Hence if you want to override the default implementation return a non-None result.
+        But in most cases you would not need to overwrite the default implementation.
+
+        Instead often it may be needed to wrap around it so that you can track the execution of the query.
+
+        For example:
+
+        .. code-block::
+
+            @hookimpl(hookwrapper=True)
+            db_connection_execute_operation(execution_cursor: ExecutionCursor) -> Optional[int]:
+                # let's track duration of the query
+                start = time.time()
+                log.info(f"Starting query: {execution_cursor.get_managed_operation().get_operation()}")
+                outcome: pluggy.callers._Result
+                outcome = yield # we yield the execution to other implementations (including default one)
+                is_success: bool = outcome.excinfo is None
+                log.info(f"Query finished. duration: {time.time() - start}. is_success: {is_success}")
+                # no return!
```

Comment:
this is a very complex simple example, how error-prone is that?

Reply:
What is so complex here - we track duration of a query ...